controllers/component_build_controller.go

```diff
@@ -142,6 +143,26 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 func (r *ComponentBuildReconciler) SubmitNewBuild(ctx context.Context, component appstudiov1alpha1.Component) error {
 	log := r.Log.WithValues("Namespace", component.Namespace, "Application", component.Spec.Application, "Component", component.Name)
 
+	// TODO delete this block which is workaround for delayed sync of pvc
+	workspaceStorage := gitops.GenerateCommonStorage(component, "appstudio")
+	existingPvc := &corev1.PersistentVolumeClaim{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: workspaceStorage.Name, Namespace: workspaceStorage.Namespace}, existingPvc); err != nil {
+		if errors.IsNotFound(err) {
+			// Patch PVC size to 1 Gi, because default 10 Mi is not enough
+			workspaceStorage.Spec.Resources.Requests["storage"] = resource.MustParse("1Gi")
+			// Create PVC (Argo CD will patch it later)
+			err = r.Client.Create(ctx, workspaceStorage)
+			if err != nil {
```

Comment:
Then within the PollImmediate, if the error here is "already exists" we just move on

Otherwise we retry, in case there are some intermittent error.

Then if things keep on failing for a few seconds, we give up and return the unable to create common storage